### PR TITLE
created config queue consumer to process config event

### PR DESCRIPTION
**Pull request by osiievska23**

### CODE REVIEW

File: src/main/java/com/icthh/xm/ms/configuration/topic/ConfigurationUpdateEventProcessor.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+@RequiredArgsConstructor
+public class ConfigurationUpdateEventProcessor {
+
+    private final TenantContextHolder tenantContextHolder;
+    private final LepManagementService lepManagementService;
+    private final ConfigurationService configurationService;
+
+    public void process(ConfigurationUpdateMessage message, String tenant) {
+        try {
+            init(tenant);
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/ms/configuration/topic/ConfigurationUpdateEventProcessor.java:27]: 1 - can we use `tenantContextHolder.getPrivilegedContext().execute(tenantKey, () ->` for run in tenant context

2 - can we use try with resources for `lepManagementService.beginThreadContext` inside tenant context

File: src/main/java/com/icthh/xm/ms/configuration/web/rest/ConfigMapResource.java [hunk 2]
```diff
@@ -62,6 +67,12 @@ public ResponseEntity<Void> updateConfiguration(@RequestBody Configuration confi
         return ResponseEntity.ok().build();
     }
 
+    private void assertUpdateConfigurationAvailable() {
+        if (!applicationProperties.isUpdateConfigAvailable()) {
+            throw new BusinessException("Configuration update is not available");
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/ms/configuration/web/rest/ConfigMapResource.java:72]: better to throw AccessDeniedException (403)

File: src/main/resources/config/application.yml [hunk 3]
```diff
@@ -150,6 +151,10 @@ application:
         max-attempts: 3
         delay: 10000 #in milliseconds
         multiplier: 2
+    config-queue-retry:
```
[the remark this answers is not in the document]
Comment (osiievska23) [reply src/main/resources/config/application.yml:154]: it's already true by default in application properties:
`private boolean updateConfigAvailable = true;`

File: src/main/java/com/icthh/xm/ms/configuration/web/rest/ConfigMapResource.java (hunk 2 above)
Comment (osiievska23) [reply src/main/java/com/icthh/xm/ms/configuration/web/rest/ConfigMapResource.java:72]: AccessDeniedException is unheadeled, is it ok?

File: src/main/java/com/icthh/xm/ms/configuration/topic/ConfigurationUpdateEventProcessor.java (hunk 1 above)
Comment (osiievska23) [reply src/main/java/com/icthh/xm/ms/configuration/topic/ConfigurationUpdateEventProcessor.java]: done

File: src/main/java/com/icthh/xm/ms/configuration/web/rest/ConfigMapResource.java (hunk 2 above)
Comment (osiievska23) [reply src/main/java/com/icthh/xm/ms/configuration/web/rest/ConfigMapResource.java]: done

File: src/main/resources/config/application.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sergeysenja1992) [reply src/main/resources/config/application.yml:154]: yes, I see, question about application.yml file

[the remark this answers is not in the document]
Comment (osiievska23) [reply src/main/resources/config/application.yml:155]: done

**Review by amedvedchuk — APPROVED** [review]

**Review by sergeysenja1992 — APPROVED** [review]

**Pull request merged**